cirq-google/cirq_google/calibration/phased_fsim.py

```diff
@@ -1038,3 +1038,35 @@ def try_convert_sqrt_iswap_to_fsim(gate: cirq.Gate) -> Optional[PhaseCalibratedF
         return PhaseCalibratedFSimGate(cirq.FSimGate(theta=np.pi / 4, phi=0.0), 0.5)
 
     return None
+
+
+def try_convert_gate_to_fsim(gate: cirq.Gate) -> Optional[PhaseCalibratedFSimGate]:
+    """Converts a gate to equivalent PhaseCalibratedFSimGate if possible.
+
+    Args:
+        gate: Gate to convert.
+
+    Returns:
+        If provided gate is equivalent to some PhaseCalibratedFSimGate, returns that gate.
+        Otherwise returns None.
+    """
+    if isinstance(gate, cirq.FSimGate):
+        return PhaseCalibratedFSimGate(gate, 0.0)
```

Comment:
If you look at the previous method, the code is trying to unify the sqrt_iswap ad sqrt_iswap inverse into a shared `engine_gate`. This could also be done here, the `PhasedFSimGate` is not an unique representation and there is a freedom between `theta` and `chi`.

It is important to do so because of the way characterizations work: they are applied to the entire moment and if we could unify two different moments into a single characterization then we'll save a lot of time. Characterizations take considerable amount of time to do.

(The same applies to IswapPowGate, CZPowGate and PhasedIswapPowGate cases).

Please reach out to me over email if you want to have some references about the math behind this. I have some notes on this in multiple places and I can summarize it for you.

Reply:
I think I understand what needs to be done here. According to my calculations, matrix of `PhaseCalibratedFSimGate(FSimGate(theta, phi), p)` is:

```
1, 0, 0, 0
0, cos(theta), exp(2*pi*p*i)*(-sin(theta)), 0
0, exp(-2*pi*p*i)*(-sin(theta)), cos(theta), 0
0, 0, 0, exp(i*phi)
```

So the transformation `theta -> -theta; p -> p+ 0.5` would not change the matrix. Therefore, by using equivalent tranformations of parameters we can always put theta in the range [0, pi).
I will do that and it will be not worse than what is done `try_convert_sqrt_iswap_to_fsim`, because sqrt(iSWAP) and its inwerse both will be mapped to the same engine gate.